crates/lib3h/src/transport/memory_mock/transport_memory.rs

```diff
@@ -115,59 +137,75 @@ impl Transport for TransportMemory {
+        trace!("TransportMemory[{}].close({})", self.own_id, id);
+        if self.maybe_my_uri.is_none() {
+            return Err(TransportError::new(
+                "Cannot close a connection before bounding".to_string(),
+            ));
+        }
+        let my_uri = self.maybe_my_uri.clone().unwrap();
         // Get the other node's uri on that connection
-        let maybe_url = self.connection_map.get(id);
-        if let None = maybe_url {
-            return Err(TransportError::new(format!(
-                "No known connection for connectionId {}",
-                id
-            )));
+        let maybe_other_uri = self.outbound_connection_map.get(id);
+        if let None = maybe_other_uri {
+            return Err(TransportError::new(format!("Unknown connectionId: {}", id)));
         }
-        let url = maybe_url.unwrap();
+        let other_uri = maybe_other_uri.unwrap();
         // Get other node's server
         let server_map = memory_server::MEMORY_SERVER_MAP.read().unwrap();
-        let maybe_server = server_map.get(url);
-        if let None = maybe_server {
+        let maybe_other_server = server_map.get(other_uri);
+        if let None = maybe_other_server {
             return Err(TransportError::new(format!(
                 "No Memory server at this url: {}",
-                url,
+                other_uri,
             )));
         }
-        let mut server = maybe_server.unwrap().lock().unwrap();
-        // Tell it to close connection
-        server.close(&id)?;
+        let mut other_server = maybe_other_server.unwrap().lock().unwrap();
+        // Tell it we closed connection with it
+        let _ = other_server.request_close(&my_uri);
         // Locally remove connection
-        self.connection_map.remove(id);
+        self.outbound_connection_map.remove(id);
+        // FIXME
+        // self.inbound_connection_map.remove(id);
+        // Done
         Ok(())
     }
 
     /// Close all known connectionIds
     fn close_all(&mut self) -> TransportResult<()> {
         let id_list = self.connection_id_list()?;
         for id in id_list {
-            self.close(&id)?;
+            let res = self.close(&id);
+            if let Err(e) = res {
+                warn!("Closing connection {} failed: {:?}", id, e);
+            }
         }
         Ok(())
     }
 
     /// Send payload to known connectionIds in `id_list`
     fn send(&mut self, id_list: &[&ConnectionIdRef], payload: &[u8]) -> TransportResult<()> {
+        if self.maybe_my_uri.is_none() {
+            return Err(TransportError::new(
+                "Cannot send before bounding".to_string(),
+            ));
+        }
+        let my_uri = self.maybe_my_uri.clone().unwrap();
         for id in id_list {
             // Get the other node's uri on that connection
-            let maybe_uri = self.connection_map.get(*id);
+            let maybe_uri = self.outbound_connection_map.get(*id);
```

Comment:
What do you think about: 
```rust
let maybe_uri = self.outbound_connection_map.get(*id).unwrap_or({
    warn!("No known connection for connectionId: {}", id);
    continue
}
```

Reply:
doesn't work in Rust like previous comment.